### PR TITLE
Allow to set the text color in WebPathField

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
Note, after my first push I amended the commit to add "rollover" to the variable to make it clearer, but it the name is not clear enough feel free to modify it.

**Comment by mgarin** [comment]
I have moved style field into separate `WebPathFieldStyle` class and I have also made use of default `WebPathField` foreground so the separate field is not required anymore.

Also foreground is now properly updated in runtime and set for both - field and buttons.

**Comment by iMichka** [comment]
Hi. Thanks for the changes. I hesitated to add a `WebPathFieldStyle` class.

Maybe I was not completely clear in the purpose of my change. In fact, I changed the background color of all buttons to a dark grey, so that now the text is black on a dark background. That's why I wanted to be able to set the color only for the rollover mode. I think there is a need for two foreground colors: normal / rollover. They are currently all set to black but I would like to set the rollover separately.

Thanks.

**Comment by mgarin** [comment]
I am afraid buttons do not support rollover foreground yet and it would be a really case-specific thing to bridge this option through the whole path ( WebButtonUI -> WebButton -> WebPathField).

Generally - I am aware that there are a lot of cases when you want to re-style components when those are used inside of other complex components. Creating tons of bridge methods is actually a really bad and temporary workaround which only make things difficult.

What is going to happen when StyleManager starts supporting such components (WebButton for instance) - they will have their default style ID (for example "button") for separate usage and will also have distinct style IDs set when used within complex components (for example "file-chooser-accept-button"). So you will be able to re-style those through style XML files in one go for separate usage cases without even touching components. You can even create your own usage cases by simply creating distinct style with custom ID.

You can browse some basics of the styling by launching `com.alee.extended.style.StyleEditor` class - this a test application with a few sample styles which I didn't yet show anywhere (though it was always inside the sources and releases) - it perfectly illustrates what `StyleManager` is capable of. For instance - it allows style editing "on fly" with instant style application to the editor itself and previewed components, for example if you try editing scrollbar settings...
image
By the way, some values could call custom editors through Ctrl+Click like on the screenshot.

Basically - StyleManager is already working within WebLaF but affects only supported components. More components support will be available in upcoming v1.29 update pretty soon (including button styling support), others will be added with the next few updates.

So for this specific case - I suggest waiting the update with button styling to re-style it through XML skin and not manually. By the way, the same styling can already be applied to labels, panels, popup menus and scroll bars.

**Comment by iMichka** [comment]
Thanks for the very detailed answer.

With the merged change and your update I can set a single foreground color, which I can live with for the moment.

I totally subscribe to the idea that adding hundreds of bridge methods is a bad idea. I really look forward to a more complete integration of the styleeditor. I already use the xml file for some label and panel styling and it is a very convenient way to do it. So I am ready for the shift to this new way of styling components.

I'll give a quick try to the StyleManager. Seems very cool.

**Comment by mgarin** [comment]
I won't lie - styling is in a bit of mess right now since half of the UI is still working in the old-fashioned way and some features are going to be changed, but I have also implemented this concept (style IDs usage) in a few projects and was really helpful and convenient. It helps a lot to get rid of separate styling for each swing component which becomes messy in large applications.

So I am certainly going to transfer the whole L&F to this new way of dealing with UI with next few updates.
